plugin/someteam.example.com/v1/replacementtransformer/ReplacementTransformer.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/kustomize/api/filters/replacement"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+// Replace values in targets with values from a source
+type plugin struct {
+	Replacements []types.Replacement `json:"replacements,omitempty" yaml:"replacements,omitempty"`
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+func (p *plugin) Config(
+	_ *resmap.PluginHelpers, c []byte) (err error) {
+	p.Replacements = []types.Replacement{}
+	err = yaml.Unmarshal(c, p)
+	if err != nil {
+		return err
+	}
+	return validateReplacement(p.Replacements)
+}
+
+func validateReplacement(repl []types.Replacement) error {
```

Comment:
Why do these validations here rather than in the filter? If someone were to use the filter directly, I'd expect we'd still want to throw errors under these circumstances, no?